client/dimensions_api.go

```diff
@@ -60,6 +60,15 @@ type DatasetAPI struct {
 	HTTPClient          HTTPClient
 }
 
+func NewDatasetAPI(host string, authToken string, responseBodyReader ResponseBodyReader, httpClient HTTPClient) DatasetAPI {
+	return DatasetAPI{
+		DatasetAPIHost:      host,
+		DatasetAPIAuthToken: authToken,
+		ResponseBodyReader:  responseBodyReader,
+		HTTPClient:          httpClient,
```

Comment:
Consider making the fields of the struct private, so they can only be set via this constructor function.

Reply:
Or remove the constructor if there's a need for them to all be public :)